web/main.py

```diff
@@ -1,109 +1,68 @@
-        if targetSplits[1].find('.') > 0:
-            newPath = "/".join(targetSplits[1:])
-            if newPath == '':
-                _serve(self.response, mainPage)
-            else:
-                _serve(self.response, newPath)
-            return
-
-        # If it is a request for a TLD psuedo-item, serve back the main page
-        if len(targetSplits) < 3:
-            _serve(self.response, mainPage)
-            return
-
-        # If it is a request for something in the packages folder, serve it
-        if targetSplits[1] == 'packages':
-            newPath = "/".join(targetSplits[1:])
-            if newPath == '':
-                _serve(self.response, mainPage)
-            else:
-                _serve(self.response, newPath)
-            return
-
-        # Otherwise it's a request for a item after the gist pseudo path
-        # drop the gist and serve it.
-        if len(targetSplits) >= 3:
-            newPath = "/".join(targetSplits[2:])
-            if newPath == '':
-                _serve(self.response, mainPage)
-            else:
-                _serve(self.response, newPath)
-            return
-
-
-# Return whether we're running in the development server or not.
-def isDevelopment():
-    return os.environ['SERVER_SOFTWARE'].startswith('Development')
-
-
-# Serve the files.
-def _serve(resp, path):
-    # Normalize and restrict the path to only files under this directory.
-    path = os.path.abspath(path)
-
-    if not path.startswith(SERVING_DIRECTORY):
-        logging.warning(
-            "Rejecting file outside of serving directory. "
-            "Serving directory: %s, requested path: %s",
-            SERVING_DIRECTORY, path)
-
-        resp.status = 404
-        resp.write("<html><h1>404: Not found</h1></html>")
-        return
-
-    if not os.path.isfile(path):
-        resp.status = 404
-        resp.write("<html><h1>404: Not found</h1></html>")
-        return
-
-    if path.endswith('.css'):
-        resp.content_type = 'text/css'
-    if path.endswith('.svg'):
-        resp.content_type = 'image/svg+xml'
-    if path.endswith('.js'):
-        resp.content_type = 'application/javascript'
-    if path.endswith('.ico'):
-        resp.content_type = 'image/x-icon'
-    if path.endswith('.html'):
-        resp.content_type = 'text/html'
-    if path.endswith('.png'):
-        resp.content_type = 'image/png'
-
-    f = open(path, 'r')
-    c = f.read()
-    resp.write(c)
-    return
-
-
-app = webapp2.WSGIApplication([
-    ('.*', MainHandler)
-], debug=False)
+import string
+
+# `entrypoint` is not defined in app.yaml, so App Engine will look for an app
+# called `app` in this file.
+app = Flask(__name__)
+
+if __name__ == '__main__':
+    # This is used when running locally only.
+    app.run(host='127.0.0.1', port=8080, debug=True)
+
+# Files that the server is allowed to serve. Additional static files are
+# served via directives in app.yaml.
+VALID_FILES= [
```

Comment:
Maybe not for something this simple, but consider frozenset() if this gets much bigger.

Reply:
Thanks for the tip!